### PR TITLE
[build][sql] Make pulsar-trino-disto self-contained

### PR DESCRIPTION
**Pull request by tisonkun**
Master issue #17137

- [x] `doc-not-needed`

### CODE REVIEW

**Comment by tisonkun** [comment]
Seems good. Let me handle all the remaining renaming things and try to prepare a PIP.

File: buildtools/src/main/resources/pulsar/checkstyle.xml [hunk 1]
```diff
@@ -39,6 +39,9 @@ page at http://checkstyle.sourceforge.net/config.html -->
         <property name="max" value="120"/>
         <property name="severity" value="error"/>
 
+        <!-- Checks only for Java files. -->
+        <property name="fileExtensions" value="java"/>
```
Comment (eolivelli) [inline buildtools/src/main/resources/pulsar/checkstyle.xml:43]: this should be a separate PR

File: distribution/server/src/assemble/bin.xml [hunk 2]
```diff
@@ -23,6 +23,7 @@
           xsi:schemaLocation="http://maven.apache.org/ASSEMBLY/2.1.0 http://maven.apache.org/xsd/assembly-2.1.0.xsd">
   <id>bin</id>
   <formats>
+    <format>dir</format>
```
Comment (eolivelli) [inline distribution/server/src/assemble/bin.xml:26]: this looks like only convenient for debugging purposes. please remove.
the Pulsar repo is already huge when you build it (more than 5GB)

**Review by eolivelli — CHANGES_REQUESTED** [review]
I overall agree with the patch.
I have left some comments about the PR style

File: buildtools/src/main/resources/pulsar/checkstyle.xml (hunk 1 above)
Comment (tisonkun) [reply buildtools/src/main/resources/pulsar/checkstyle.xml:43]: OK. I include this change in the patch because after I move config files under the module, this rule takes effect and reports failure. I can do it in a separate PR though.

File: distribution/server/src/assemble/bin.xml (hunk 2 above)
Comment (tisonkun) [reply distribution/server/src/assemble/bin.xml:26]: Yep. The generated folder takes 421M. I will revert it.

File: buildtools/src/main/resources/pulsar/checkstyle.xml (hunk 1 above)
Comment (tisonkun) [reply buildtools/src/main/resources/pulsar/checkstyle.xml:43]: See https://github.com/apache/pulsar/pull/17168.

File: distribution/server/src/assemble/bin.xml (hunk 2 above)
Comment (tisonkun) [reply distribution/server/src/assemble/bin.xml]: Reverted. Also, finish `lib/presto` to `trino` migration and test locally with examples. Don't run all integration tests though.

File: bin/pulsar [hunk 3]
```diff
@@ -257,8 +257,8 @@ if [ -z "$PULSAR_LOG_CONF" ]; then
     PULSAR_LOG_CONF=$DEFAULT_LOG_CONF
 fi
 
-if [ -z "$PULSAR_PRESTO_CONF" ]; then
-    PULSAR_PRESTO_CONF=$DEFAULT_PULSAR_PRESTO_CONF
+if [ -z "$PULSAR_TRINO_CONF" ]; then
```
Comment (nicoloboschi) [inline bin/pulsar:260]: for better handling retrocompatibility we could still check PULSAR_PRESTO_CONF as fallback. something like this:

```
if [ -z "$PULSAR_TRINO_CONF" ]; then
  if [ -z "$PULSAR_PRESTO_CONF" ]; then
     PULSAR_TRINO_CONF=$DEFAULT_PULSAR_TRINO_CONF
   else
     PULSAR_TRINO_CONF=$PULSAR_PRESTO_CONF
   fi
fi
```

Comment (tisonkun) [reply bin/pulsar:266]: Implemented in 43c8c7da41154289baf4fd26e568ce2ace4ae451.

**Review by nicoloboschi — APPROVED** [review]

File: bin/pulsar (hunk 3 above)
Comment (tisonkun) [reply bin/pulsar:266]: Oops. I think I misunderstand your suggestion, followup as 362207d3e54369175bca2553609f10ba0855ca7d.

Then a version upgrade should go well, while a downgrade needs a caveat. And anyway, PIP-200 publicly drop the previous configure so that we can drop it in the following versions. This is for a smoothly deprecating period.

File: bin/pulsar [hunk 4]
```diff
@@ -177,17 +177,23 @@ if [ ! -f "${PY_INSTANCE_FILE}" ]; then
...
-        BUILT_PRESTO_HOME="${SQL_HOME}/presto-distribution/target/pulsar-presto-distribution"
-        if [ ! -d "${BUILT_PRESTO_HOME}" ]; then
-            echo "\nCouldn't find presto distribution.";
-            echo "Make sure you've run 'mvn package'\n";
-            exit 1;
+# find pulsar sql trino distribution location
+check_trino_libraries() {
+    if [ ! -d "${TRINO_HOME}" ]; then
+        # TODO: As PIP-200 accepted, this compatibility is not promised. Refactor when we drop this b/w compatibility.
+        FALLBACK_PRESTO_HOME="${PULSAR_HOME}/lib/presto"
```
Comment (tisonkun) [inline bin/pulsar:184]: This should be internal dependency and users should not and cannot manage it.

Pulsar SQL tightly couples with the bundled trino distro.

Will revert the "compatibility" part.

Comment (tisonkun) [reply bin/pulsar:184]: Concretely, <= 2.11 works with prestosql 334 or 333 under lib/presto, >= 2.12 works with trino 353 with breaking changes.

By no means one uses 2.12 wants to "fallback" to the previous lib folder.

Comment (tisonkun) [reply bin/pulsar]: Revert at 54fb61e330d2bc963234727e53b3440c651906ae

**Comment by tisonkun** [comment]
ping @codelipenghui @eolivelli @maxsxu @nodece @lhotari as you voted for this PIP >_<

File: .github/workflows/ci-owasp-dep-check.yaml [hunk 5]
```diff
@@ -87,7 +87,7 @@ jobs:
           docker rmi $(docker images -q) -f
           df -h
 
-      # Projects dependent on flume, hdfs, hbase, and presto currently excluded from the scan.
+      # Projects dependent on flume, hdfs, hbase, and trino currently excluded from the scan.
```
Comment (tisonkun) [inline .github/workflows/ci-owasp-dep-check.yaml:90]: This patch doesn't actually touch dependencies. OWASP failed due to this comment edition triggering validation. The failures can be resolved in another turn.

**Review by codelipenghui — APPROVED** [review]
Nice work.

LGTM, I have tried on my laptop. Pulsar SQL works great.

**Review by eolivelli — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks for your reviews! I think this patch is ready for merge. @eolivelli @codelipenghui @nicoloboschi could you help with performing the merge?

After it's merged I'll update the corresponding docs and then we can close PIP-200 as completed.

**Review by maxsxu — APPROVED** [review]

**Comment by tisonkun** [comment]
It seems broker group 2 tests don't like me, lol. Update branch and rerun.

**Comment by tisonkun** [comment]
Now broker group 1 tests don't like me :(

Rerun later >_<

**Review by nodece — APPROVED** [review]

**Review by MarvinCai — APPROVED** [review]

**Comment by tisonkun** [comment]
@nicoloboschi I think the unrequired check status is due to your action. Do you think it's good to merge this PR now?

**Comment by tisonkun** [comment]
Finally!

@codelipenghui @nicoloboschi could you help with merging this XD
